jdbc/src/main/java/org/apache/metamodel/jdbc/dialects/AbstractQueryRewriter.java

```diff
@@ -460,4 +460,26 @@ public Object getResultSetValue(ResultSet resultSet, int columnIndex, Column col
         }
         return resultSet.getObject(columnIndex);
     }
+
+    protected boolean isSupportedVersion(String databaseProductName, int databaseVersion) {
+
+        if(databaseProductName.equals(_dataContext.getDatabaseProductName())
+                && databaseVersion <= getDatabaseMajorVersion(_dataContext.getDatabaseVersion())) {
+            return true;
+        }
+        return false;
+    }
+
+    private int getDatabaseMajorVersion(String version) {
+        int firstDot = -1;
+        if(version != null) {
+            version = version.replaceAll("[^0-9.]+", "");
+            firstDot = version.indexOf('.');
+        }
+        if(firstDot >= 0) {
+            return Integer.valueOf(version.substring(0, firstDot));
```

Comment:
Not sure if we want to make this more "defensive" in terms of programming. I can't imagine it really, but if the string argument is just "." then this substring will return empty string and that will throw exception when parsing as integer. It's not really something I think is likely to happen, but just saying :-)